src/main/java/org/terasology/ReplayTestingEnvironment.java

```diff
@@ -168,6 +204,13 @@ protected boolean isInitialised() {
         return this.isInitialised;
     }
 
-
-
+    protected void waitUntil(BooleanSupplier supplier) {
+        try {
+            while (!supplier.getAsBoolean()) {
+                Thread.sleep(1000);
```

Comment:
I still feel this should be changed to a smaller value, like `100` (10 FPS) or `50` (20 FPS). Ideally I'd wait for 16 ms (60 FPS). I think it is worth a try, and I don't see why sleeping for a shorter period of time should decrease performance.